### PR TITLE
📚 Update contribution guidelines 

### PR DESCRIPTION
**Pull request by samet-akcay**
## 📝 Description

- Update `CONTRIBUTING.md` file
- Add `Developer Guide` to the documenation.

Currently developer guide only has the contribution guidelines, but we could add more sections later on.

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -26,50 +26,98 @@ We utilize GitHub issues to track the feature requests as well. If you are certa
...
 
-```bash
-conda create -n anomalib_dev python=3.10
-conda activate anomalib_dev
-pip install -r requirements/base.txt -r requirements/dev.txt
-pre-commit install
-```
+#### 2. Set Up Your Development Environment
+
+Set up your development environment to start contributing. This involves installing the required dependencies and setting up pre-commit hooks for code quality checks.
```
Comment (djdameln) [inline CONTRIBUTING.md:37]: For completeness we could add here that this guide will use conda (and assumes the developer has conda installed on their system), but the steps are the same for other env management tools.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -26,50 +26,98 @@ We utilize GitHub issues to track the feature requests as well. If you are certa
...
+   pip install -e .[all]
+   ```
+
+3. Install and configure pre-commit hooks:
+
+   ```bash
+   pre-commit install
+   ```
+
+Pre-commit hooks help ensure code quality and consistency. To manually run pre-commit checks:
```
Comment (djdameln) [inline CONTRIBUTING.md:67]: Should we explain here that pre-commit runs automatically before each commit?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -26,50 +26,98 @@ We utilize GitHub issues to track the feature requests as well. If you are certa
...
+   For more on signing commits, see [GitHub's guide on signing commits](https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/signing-commits).
+
+### Submitting Pull Requests
+
+Once you've followed the above steps and are satisfied with your changes:
+
+1. Push your changes to your forked repository.
+2. Go to the original Anomalib repository you forked and click "New pull request".
+3. Choose your fork and the branch with your changes to open a pull request.
+4. Fill in the pull request template with the necessary details about your changes.
```
Comment (djdameln) [inline CONTRIBUTING.md:117]: 5. Update changelog?

Comment (samet-akcay) [reply CONTRIBUTING.md:117]: It is in `Make Changes` section?

**Review by djdameln — APPROVED** [review]

**Pull request merged**